.buildkite/dockerfiles-pipeline.yml

```diff
@@ -0,0 +1,124 @@
+        artifact_paths: ".artifacts/*.tar.gz"
+      - label: "Test amd64 image built from Dockerfile"
+        agents:
+          provider: aws
+          instanceType: m6i.xlarge
+          imagePrefix: ci-amazonlinux-2
+        env:
+          ARCHITECTURE: "amd64"
+          DOCKERFILE_PATH: "Dockerfile"
+          DOCKER_IMAGE_NAME: "docker.elastic.co/ci-agent-images/elastic-connectors-dockerfile"
+          DOCKER_ARTIFACT_KEY: "elastic-connectors-dockerfile"
+        depends_on: "build_dockerfile_image_amd64"
+        key: "test_dockerfile_image_amd64"
+        commands:
+          - "mkdir -p .artifacts"
+          - buildkite-agent artifact download '.artifacts/*.tar.gz*' .artifacts/ --step build_dockerfile_image_amd64
+          - ".buildkite/publish/test-docker.sh"
+
+      # ----
+      # Dockerfile.ftest build and tests on amd64
+      # ----
+      - label: "Build amd64 image from Dockerfile.ftest"
+        agents:
+          provider: aws
+          instanceType: m6i.xlarge
+          imagePrefix: ci-amazonlinux-2
+        env:
+          ARCHITECTURE: "amd64"
+          DOCKERFILE_PATH: "Dockerfile.ftest"
+          DOCKER_IMAGE_NAME: "docker.elastic.co/ci-agent-images/elastic-connectors-dockerfile-ftest"
+          DOCKER_ARTIFACT_KEY: "elastic-connectors-dockerfile-ftest"
+        command: ".buildkite/publish/build-docker.sh"
+        key: "build_dockerfile_ftest_image_amd64"
+        artifact_paths: ".artifacts/*.tar.gz"
+      - label: "Test amd64 image built from Dockerfile.ftest"
+        agents:
+          provider: aws
+          instanceType: m6i.xlarge
+          imagePrefix: ci-amazonlinux-2
+        env:
+          ARCHITECTURE: "amd64"
+          DOCKERFILE_PATH: "Dockerfile.ftest"
+          DOCKER_IMAGE_NAME: "docker.elastic.co/ci-agent-images/elastic-connectors-dockerfile-ftest"
+          DOCKER_ARTIFACT_KEY: "elastic-connectors-dockerfile-ftest"
+        depends_on: "build_dockerfile_ftest_image_amd64"
+        key: "test_dockerfile_ftest_image_amd64"
+        commands:
+          - "mkdir -p .artifacts"
+          - buildkite-agent artifact download '.artifacts/*.tar.gz*' .artifacts/ --step build_dockerfile_ftest_image_amd64
+          - ".buildkite/publish/test-docker.sh"
+
+
+      # ----
+      # Vulnerability scanning on Dockerfile and Dockerfile.ftest built images
+      # ----
+      - label: "Trivy Scan amd64 Dockerfile Artifacts"
+        timeout_in_minutes: 10
+        depends_on:
+          - test_dockerfile_image_amd64
+        key: "trivy-scan-dockerfile-image"
+        agents:
+          provider: k8s
+          image: "docker.elastic.co/ci-agent-images/trivy:latest"
+        command: |-
+          mkdir -p .artifacts
+          buildkite-agent artifact download '.artifacts/*.tar.gz*' .artifacts/ --step build_dockerfile_image_amd64
+          trivy --version
+          env | grep TRIVY
+          find .artifacts -type f -name '*.tar.gz*' -exec trivy image --quiet --input {} \;
```

Comment:
nice 👍 

I'm thinking it could also be nice to add one additional stage to this pipeline, if it's not too difficult:

a smoke-test/acceptance-test that verifies that connectors actually starts up in the resultant image, just as a way of gaining confidence that the built image does what we expect it to do

Reply:
Good point, this pipeline includes the same test step that is used for the production docker images built from Dockerfile.wolfi [here](https://github.com/elastic/connectors/blob/kostasb/switch-dockerfile-image/.buildkite/pipeline.yml#L637) so that the testing approach is aligned on both built images. 

The pipeline step `test_extensible_dockerfile_image_amd64` runs the [.buildkite/publish/test-docker.sh](https://github.com/elastic/connectors/blob/kostasb/switch-dockerfile-image/.buildkite/extensible-dockerfiles-pipeline.yml#L37C14-L37C47) which performs basic sanity checks and verifies the output of `elastic-ingest --version` [here](https://github.com/elastic/connectors/blob/kostasb/switch-dockerfile-image/.buildkite/publish/test-docker.sh#L79).

Since adding functional tests at the connector level is a more involved task which could benefit both the production and the "extensible" image I'd suggest we create a dedicated issue for that, what do you think?